src/lib/UtapiClient.js

```diff
@@ -335,6 +338,74 @@ export default class UtapiClient {
     }
 
     /**
+    * Updates counter for the putDeleteMarkerObject action
+    * @param {object} params - params for the metrics
+    * @param {string} [params.bucket] - (optional) bucket name
+    * @param {string} [params.accountId] - (optional) account ID
+    * @param {string} [params.userId] - (optional) user ID
+    * @param {number} timestamp - normalized timestamp of current time
+    * @param {string} action - action metric to update
+    * @param {object} log - Werelogs request logger
+    * @param {function} cb - callback to call
+    * @return {undefined}
+    */
+    _pushMetricDeleteMarkerObject(params, timestamp, action, log, cb) {
+        this._checkProperties(params);
+        this._logMetric(params, '_pushMetricDeleteMarkerObject', timestamp,
+            log);
+        const cmds = [];
+        const paramsArr = this._getParamsArr(params);
+        paramsArr.forEach(p => cmds.push(
+            ['incr', generateCounter(p, 'numberOfObjectsCounter')],
+            ['incr', generateKey(p, 'deleteObject', timestamp)]));
+        // We track the number of commands needed for each `paramsArr` property
+        // to eventually locate each group in the results from Redis.
+        const commandsGroupSize = 2;
+        return this.ds.batch(cmds, (err, results) => {
+            if (err) {
+                log.error('error pushing metric', {
+                    method: 'UtapiClient._pushMetricDeleteMarkerObject',
+                    error: err,
+                });
+                return this._pushLocalCache(params, action, timestamp, log, cb);
+            }
+            let actionErr;
+            let actionCounter;
+            const cmds2 = [];
+            const noErr = paramsArr.every((p, i) => {
+                // We want the first element of every group of two commands
+                // returned from Redis. This contains the value of the
+                // numberOfObjectsCounter after it has been incremented.
+                const index = i * commandsGroupSize;
+                actionErr = results[index][0];
+                actionCounter = parseInt(results[index][1], 10);
+                // If < 0 or NaN, record numberOfObjects as though bucket were
+                // empty.
+                actionCounter = Number.isNaN(actionCounter) ||
+                    actionCounter < 0 ? 1 : actionCounter;
```

Comment:
I noticed in S3 we put a delete marker whether a marker exists or not. Is this correct? If so, there could be a case where the bucket is empty yet a delete marker is put.